docs/developers/v2/hacking-with-yearn.md

```diff
@@ -0,0 +1,237 @@
+# Hacking on Yearn
+
+This page is for the hackers and buidlers who are currently building, or planning to build, something cool with Yearn Finance. This guide  details all the info you need to start hacking!
+
+## Contents
+- Vault and Smart Contract Integrations
+- Front-end and SDK Integrations
+- Strategy Development
+- Data Analysis
+- Inspiration
+- Examples
+- Expectations
+
+# Core Tools
+
+- [Solidity](https://https://docs.soliditylang.org/en/v0.8.6/bugs.html)
+- [Brownie](https://eth-brownie.readthedocs.io/en/stable/)
+- [ethers.js](https://docs.ethers.io/v5/)
+- [Web3.js](https://web3js.readthedocs.io/en/v1.4.0/)
+- [Ganache](https://trufflesuite.com/ganache/) 
+- [Foundry](https://github.com/nicolasgarcia214/damn-vulnerable-defi-foundry)
+
+# Yearn Vaults & Smart Contract Integrations
+Yearn’s core product is the Vault (also known as yVault) — a set-it-and-forget-it yield aggregator running on top of DeFi’s yield-generating protocols like Compound, Aave, Curve, and Convex. Most of the core functionality such as deposit, withdraw, borrow, repay, or flashloan occurs in the XX contract ().
+For a breakdown of V2 features see the contract overview or dive even deeper with the V2 Technical Paper.
+
+## Getting Started
+
+#### Starter Kit
+#### 
+Here's an example for integrating V2 vaults and ERC-4626. 
+
+https://github.com/storming0x/ystarter-foundry-kit
+
+What you'll find:
+- Basic example Solidity Smart Contracts for integrating with Yearn Vaults for both V2 and ERC4626 interfaces.
+- ERC4626 adapter to wrap yearn vaults.
+- Example Contracts
+- Sample test suite.
+
+
+Yearn Vaults
+
+[This repository](https://github.com/yearn/yearn-vaults#readme) includes the set of smart contracts that are used for the Yearn Vaults. It contains the requirements, code, deployment scripts, and tests necessary for the core protocol, including a inheritable template strategy for use with Solidity-based strategies that interact with Yearn Vaults. These contracts are used to create a simple way to generate high risk-adjusted returns for depositors of various assets via best- in-class lending protocols, liquidity pools, and community-made yield farming strategies on Ethereum.
+
+
+#### 
+## Testing
+
+### **Introduction**
```

Comment:
i don't think that a title with ** ** is a good idea. Let's keep the styling clean